libs/langchain/langchain/chat_models/huggingface_llama2.py

```diff
@@ -0,0 +1,188 @@
+import importlib.util
+from enum import Enum
+from typing import Any, Dict, List, Optional, Union
+
+from langchain.callbacks.manager import (
+    CallbackManagerForLLMRun,
+)
+from langchain.chat_models.base import BaseChatModel
+from langchain.pydantic_v1 import root_validator
+from langchain.schema import ChatResult
+from langchain.schema.messages import (
+    AIMessage,
+    BaseMessage,
+    HumanMessage,
+    SystemMessage,
+)
+from langchain.schema.output import ChatGeneration
+
+
+class InstructionTokens(Enum):
+    def __str__(self) -> str:
+        return self.value
+
+    B_INST = "[INST]"
+    E_INST = "[/INST]"
+
+
+class SystemTokens(Enum):
+    def __str__(self) -> str:
+        return self.value
+
+    B_SYS = "<<SYS>>"
+    E_SYS = "<</SYS>>"
+
+
+class ChatLlama2Hf(BaseChatModel):
+    pipeline: Any
+
+    @property
+    def _llm_type(self) -> str:
+        """Return type of chat model."""
+        return "llama-2-chat-hf"
+
+    @root_validator(pre=True)
+    def validate_environment(cls, values: Dict) -> Dict:
+        if (
+            not hasattr(values["pipeline"], "task")
+            or values["pipeline"].task != "text-generation"
+        ):
+            raise ValueError("The pipeline task should be 'text-generation'.")
+
+        return values
+
+    @staticmethod
+    def format_messages_as_text(messages: List[BaseMessage]) -> str:
+        """
+        Transform List of Chat Messages to text following Meta's prompt guidelines.
+
+        Prompt template with System Message:
+        ```
+        <s>[INST] <<SYS>>
```

Comment:
Interesting, this probably means we need to update:

https://github.com/langchain-ai/langchain/blob/master/libs/langchain/langchain/chat_models/ollama.py

Reply:
Managed to track down a source but might be nice to include this in the docs/in a comment here?

https://github.com/facebookresearch/llama-recipes/blob/df77625e48c3994aef19702fb331215f7fb83494/docs/inference.md?plain=1#L124

Reply:
Yes, I think so. I'll include in the docstring